packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,373 @@
+  onPopoverClose
+} from './SearchPanel/SearchPanelUtils'
+import { getBridgeUiConfigForChain } from '../../util/bridgeUiConfig'
+import { getWagmiChain } from '../../util/wagmi/getWagmiChain'
+import { useNetworks } from '../../hooks/useNetworks'
+
+type NetworkType = 'core' | 'orbit'
+
+const chainGroupInfo: {
+  [key in NetworkType]: {
+    name: string
+    description: string
+  }
+} = {
+  core: {
+    name: 'CORE CHAINS',
+    description: 'Chains managed directly by Ethereum or Arbitrum'
+  },
+  orbit: {
+    name: 'ORBIT CHAINS',
+    description: 'Independent projects using Arbitrum technology.'
+  }
+}
+
+function NetworkRow({
+  chainId,
+  style,
+  onClick,
+  close
+}: {
+  chainId: ChainId
+  style: CSSProperties
+  onClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const { network, description, chainType, nativeTokenData } =
+    getBridgeUiConfigForChain(chainId)
+  const chain = getWagmiChain(chainId)
+  const [{ sourceChain }] = useNetworks()
+
+  function handleClick() {
+    onClick(chain)
+    close() // close the popover after option-click
+  }
+
+  function onKeyDown(event: React.KeyboardEvent<HTMLButtonElement>) {
+    if (event.key === 'Enter') {
+      handleClick()
+    }
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      onKeyDown={onKeyDown}
+      key={chainId}
+      style={style}
+      type="button"
+      aria-label={`Switch to ${network.name}`}
+      className={twMerge(
+        'flex h-[90px] w-full items-center gap-4 px-6 py-2 text-lg hover:bg-black/10',
+        chainId === sourceChain.id && 'bg-black/10' // selected row
+      )}
+    >
+      <span className="flex h-6 w-6 shrink-0 items-center justify-center lg:h-6 lg:w-6">
+        <Image
+          src={network.logo}
+          alt={`${network.name} logo`}
+          className="h-full w-auto"
+          width={24}
+          height={24}
+        />
+      </span>
+      <div className={twMerge('flex flex-col items-start gap-1')}>
+        <span className="truncate leading-none">{network.name}</span>
+        {description && (
+          <p className="whitespace-pre-wrap text-left text-xs leading-[1.15]">
+            {description}
+          </p>
+        )}
+        {chainType && (
+          <p className="text-[10px] leading-none">
+            {chainType} Chain, {nativeTokenData?.symbol ?? 'ETH'} is the native
+            gas token
+          </p>
+        )}
+      </div>
+    </button>
+  )
+}
+
+function AddCustomOrbitChainButton() {
+  const [, setQueryParams] = useArbQueryParams()
+  const { isTestnetMode } = useIsTestnetMode()
+
+  if (!isTestnetMode) {
+    return null
+  }
+
+  return (
+    <button
+      className="text-sm underline"
+      onClick={() => setQueryParams({ settingsOpen: true })}
+    >
+      <span>Add Custom Orbit Chain</span>
+    </button>
+  )
+}
+
+function NetworksPanel({
+  networks,
+  onNetworkRowClick,
+  close
+}: {
+  networks: ChainId[]
```

Comment:
nit: maybe this could be something like `networkIds`, otherwise later on we do things like:
```typescript
getBridgeUiConfigForChain(network).network.name
```
`(network).network` is a bit confusing, not immediately sure which one relates to the actual network/chainId